### PR TITLE
Add Client + tidy

### PR DESCRIPTION
**Pull request by inverse**
First up thanks for providing this library - it was super useful to get going.

It felt like the interface for sending messages could be improved as currently the implementation relies on the message being the object dispatching the message which feels like it turning into a god object.

With this change I introduced a `Client` class to take care of that which takes a `Message` and dispatches it to the provided server.

While still maintaining backwards compatibility which could be dropped in v4.

Additionally

- use short array syntax - min PHP version set to 8
- tidy missing access modifiers

### CODE REVIEW

File: src/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Ntfy;
+
+use stdClass;
+use Ntfy\Exception\NtfyException;
+
+class Dispatcher
```
Comment (inverse) [inline src/Dispatcher.php:8]: Could also rename it to `Ntfy` if you think that's more fitting to the way you had the library before.

**Comment by VerifiedJoseph** [comment]
Thanks for the pr. this is good idea. 

I'm wondering how to handle authentication with theses changes. Currently authentication is set with the `Message` class method [`auth()`](https://github.com/VerifiedJoseph/ntfy-php-library/blob/main/src/Message.php#L230), might need to have a class just for authentication, called `Auth` or something. The username and password from `Auth` class would then be passed into `Dispatcher` via its constructor.

**Comment by inverse** [comment]
Thanks for the feedback - Created a simple value object to represent the auth and passed it through.

**Comment by VerifiedJoseph** [comment]
> Thanks for the feedback - Created a simple value object to represent the auth and passed it through.

Looks good, great work.

File: src/Dispatcher.php (hunk 1 above)
Comment (inverse) [reply src/Dispatcher.php:8]: Could also be `Client` as it's the one sending the message :)

File: tests/TestCase.php [hunk 2]
```diff
@@ -44,4 +46,22 @@ protected static function getHttpBinUri(): string
 
 		return self::$httpBinUri;
 	}
+
+	/**
+	 * Returns example message values loaded from message.json
+	 */
+	protected static function getMessageExample(): stdClass
+	{
+		$data = (string) file_get_contents(__DIR__ . '/TestAssets/message.json');
```
Comment (inverse) [inline tests/TestCase.php:55]: Could include new helper function to ensure errors are raised to make it easier to debug for missing fixtures.

```
    protected static function loadFixture(string $name): string
    {
        $fixturePath = __DIR__.'/Fixtures/'.$name;
        $contents = file_get_contents($fixturePath);

        if (false === $contents) {
            throw new RuntimeException(sprintf('Unable to load fixture: %s', $fixturePath));
        }

        return $contents;
    }
```

Comment (VerifiedJoseph) [reply tests/TestCase.php]: done. much better than having two methods.

**Pull request merged**